CMakeLists.txt

```diff
@@ -1,77 +1,48 @@
-project(json-schema-validator
+project(nlohmann_json_schema_validator
     LANGUAGES CXX)
 
-set(PROJECT_VERSION 2.0.0)
+set(PROJECT_VERSION 2.1.0)
 
 cmake_minimum_required(VERSION 3.2)
 
 option(BUILD_TESTS      "Build tests"    ON)
 option(BUILD_EXAMPLES   "Build examples" ON)
 
-# if used as a subdirectory just define a json-hpp-target as add_library(json-hpp INTERFACE)
-# and associate the path to json.hpp via target_include_directories()
-if(NOT TARGET json-hpp)
-    set(NLOHMANN_JSON_DIR "" CACHE STRING "path to json.hpp")
-
-    # find nlohmann's json.hpp
-    find_path(JSON_HPP nlohmann/json.hpp
-        PATHS
-            ${NLOHMANN_JSON_DIR}
-            ${CMAKE_BINARY_DIR}/${NLOHMANN_JSON_DIR}) # in case it is a relative path
-
-    # get the full, real path
-    get_filename_component(NLOHMANN_JSON_REALPATH ${JSON_HPP} REALPATH)
-
-    if(NOT EXISTS ${NLOHMANN_JSON_REALPATH}/nlohmann/json.hpp)
-        message(FATAL_ERROR "please set NLOHMANN_JSON_DIR to a path in which NLohmann's json.hpp can be found. Looking for nlohmann/json.hpp in '${NLOHMANN_JSON_REALPATH}")
-    endif()
-
-    # create an interface-library for simple cmake-linking
-    add_library(json-hpp INTERFACE)
-    target_include_directories(json-hpp
-        INTERFACE
-            ${NLOHMANN_JSON_REALPATH})
-endif()
+find_package(nlohmann_json REQUIRED)
```

Comment:
```suggestion
if(NOT TARGET nlohmann_json::nlohmann_json)
  find_package(nlohmann_json REQUIRED)
endif()
```

Reply:
If json-schema-validator is included as a subdirectory in another project it is likely that nlohmann_json will also be included as a subdirectory (this is the case for my project). 

In which case nlohmann_json will not be already installed and find_package will not work. 

But as long as nlohmann_json is included as a subdirectory before json-schema-validator the nlohmann_json::nlohmann_json target should already be created.

I suggest only trying to find nlohmann_json if the nlohmann_json::nlohmann_json target does not already exist.

Reply:
Where would the `nlohmann_json::nlohmann_json` come from?

Reply:
So in the case of my project I include nlohmann_json as a subdirectory, which means the `nlohmann_json::nlohmann_json` target is created as part of my project just like any other library.

Reply:
Similarly when json-schema-validator is included as a subdirectory to a project the `json-schema-validator` library is built with that project.

Reply:
So we have 3 possibilities (when using CMake)

1. using nlohmann-json's git-repo as a cmake-sub-dir
2. using nlohmann-json's installed CMake-package
3. using nlohmann's json.hpp directly - by creating an interface target pointing its `target_include_directories()` to the path where `nlohmann/json.hpp` is located.

Anything else?

If not, I suggest to force users of case 3 to create a `nlohmann_json::nlohmann_json` target and for case 2 to create a `nlohmann_json::nlohmann_json`-target in this CMakeLists.txt target-linking to `nlohmann_json` from the package. And updating the doc accordingly.
Would you agree?

Reply:
I think that covers all the options. If you use `find_package(nlohmann_json)` then `nlohmann_json::nlohmann_json` is automatically created. Besides that I agree.

Reply:
Cool, that simplifies it.